File: 1-Two Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        if len(nums) <= 1:
            return False
        buff_dict = {} #initial a blank dictionary
        for i in range(len(nums)): # get the length
            if nums[i] in buff_dict: # if found the element, then we can get the index
                return [buff_dict[nums[i]], i]
            else:
                buff_dict[target - nums[i]] = i #save the element needed with index - i

File: 1-Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hash;
	    vector<int> result;
	    for (int i = 0; i < nums.size(); i++) {
		    int numberToFind = target - nums[i]; //get the number needed

            //If I can find the number in hash
		    if (hash.find(numberToFind) != hash.end()) {
                //Then, save the index of original index and current index
			    result.push_back(hash[numberToFind]);
			    result.push_back(i);
			    //return the list			
			    return result;
		    }

            //if the number is in the hash, then save the index
		    hash[nums[i]] = i;
	    }
	    return result;
        
    }
};
